packages/components/src/components/data-list/data-list.tsx

```diff
@@ -0,0 +1,244 @@
+import classNames from 'classnames';
+import React from 'react';
+import { TransitionGroup } from 'react-transition-group';
+import {
+    List as _List,
+    CellMeasurer as _CellMeasurer,
+    CellMeasurerCache,
+    CellMeasurerProps,
+    AutoSizer as _AutoSizer,
+    type AutoSizerProps,
+    ScrollParams,
+    ListProps,
+    ListRowProps,
+} from 'react-virtualized';
+import { isMobile, isDesktop } from '@deriv/shared';
+import DataListCell from './data-list-cell';
+import DataListRow from './data-list-row';
+import ThemedScrollbars from '../themed-scrollbars';
+import { MeasuredCellParent } from 'react-virtualized/dist/es/CellMeasurer';
+
+const List = _List as unknown as React.FC<ListProps>;
+const AutoSizer = _AutoSizer as unknown as React.FC<AutoSizerProps>;
+const CellMeasurer = _CellMeasurer as unknown as React.FC<CellMeasurerProps>;
+type CellType = { Cell: typeof DataListCell };
+export type TRowRenderer = (params: { row: any; is_footer?: boolean; measure?: () => void }) => React.ReactNode;
+type DataListProps = {
+    className?: string;
+    data_source: any;
+    footer?: React.ReactNode;
+    getRowAction?: (row: React.ReactNode) => string;
+    getRowSize?: (params: any) => number;
+    keyMapper?: (row: any) => string;
+    onRowsRendered?: () => void;
+    onScroll?: (ev: ScrollParams) => void;
+    passthrough?: {
+        root_store: any;
```

Comment:
```suggestion
        root_store: TRootStore;
```
import { TRootStore } from '../types';

Reply:
as far as i checked it's implemented in other workspaces(packages) and there's no `Types` in Components workspace!
Are you sure about this?

Reply:
You can use `TStores` from `@deriv/stores` package.

```ts
import type { TStores } from '@deriv/stores';
```

But why this component needs to know about the stores and state management? Aren’t we just rendering a list of data? 😕